### PR TITLE
Refactor: Playbook and SQL loading interface

### PR DESCRIPTION
**Pull request by lritter**
The `PlaybookProvider` and `SQLProvider` interfaces
abstractthe details of loading playbooks and SQL. Doing
so helps keep implementation details regarding, for
instance, Consul separate from the logic of running steps.

These interfaces will also make it easier to add new
sources for sql and playbooks.

### CODE REVIEW

File: playbook/playbook.go [hunk 1]
```diff
@@ -39,15 +39,15 @@ func NewPlaybook() Playbook {
...
-}
+// // Dispatch to format-specific parser
+// func ParsePlaybook(playbookPath string, consulAddress string, variables map[string]string) (Playbook, error) {
+// 	// TODO: Add TOML support?
+// 	playbook, err := getAndParsePlaybookYaml(playbookPath, consulAddress)
+// 	if err == nil {
+// 		playbook = mergeCLIVariables(playbook, variables)
+// 	}
+// 	return playbook, err
+// }
```
Comment (alexanderdean) [inline playbook/playbook.go]: Let's delete this rather than comment out...

File: playbook/playbook.go [hunk 2]
```diff
@@ -39,15 +39,15 @@ func NewPlaybook() Playbook {
...
 
 func mergeCLIVariables(playbook Playbook, variables map[string]string) Playbook {
```
Comment (alexanderdean) [inline playbook/playbook.go]: Where is the merging of CLI variables performed post-refactor?

**Comment by alexanderdean** [comment]
Thanks @lritter ! Assigning to @jbeemster to review. @lritter has signed the CLA.

**Comment by lritter** [comment]
What's the next step on this? I ran some of the stuff in the integration test folder but I don't have Consul and the lack of unit tests makes it a bit difficult to verify my code myself. If there is a documented test regime that I can follow that would be swell.

**Comment by jbeemster** [comment]
Hey @lritter the PR looks really good!  Writing a comprehensive test suite is something that we really need to add soon. I will be adding tests much like those in the Snowplow Golang Tracker which should make all of this testing much easier!

File: main.go [hunk 3]
```diff
@@ -37,12 +38,27 @@ func main() {
 
 	options := processFlags()
 
-	pb, err := playbook.ParsePlaybook(options.playbook, options.consul, options.variables)
+	// pb, err := playbook.ParsePlaybook(options.playbook, options.consul, options.variables)
```
Comment (jbeemster) [inline main.go]: This can be removed rather than commented out.

File: run/db.go [hunk 4]
```diff
@@ -67,14 +62,8 @@ func prepareQuery(queryPath string, consulAddress string, template bool, variabl
...
-// Reads the file ready for executing
-func readScript(file string) (string, error) {
-
-	scriptBytes, err := ioutil.ReadFile(file)
-	if err != nil {
-		return "", err
-	}
-	return string(scriptBytes), nil
+func readScript(f string) (string, error) {
+	return "", nil
```
Comment (jbeemster) [inline run/db.go]: Can this function be removed now?  Does not appear to be doing anything anymore...

**Comment by lritter** [comment]
Hi. Can this be merged?

**Comment by alexanderdean** [comment]
Hey @lritter - it is scheduled for the next SQL Runner release. We have a bit of a backlog to clear but yes we hope to get to this soon.

**Pull request closed without merging**